Consolidated Drupal 9 Compatibility

japerry opened the pull request.

japerry committed:
Making acm_sku D9 ready/compatible.

# Conflicts:
#	modules/acm_sku/src/Entity/SKU.php

malachy-mcconnell commented on modules/acm_sku/acm_sku.install:
```diff
@@ -109,43 +113,22 @@ function acm_sku_update_8001() {
-  catch (\Exception $e) {
-    \Drupal::logger('acm_sku')->error($e->getMessage());
-  }
+  $storage_definition = BaseFieldDefinition::create('boolean')
+    ->setLabel(t('Status'))
+    ->setDescription(t('Whether the SKU is available or not.'))
+    ->setDefaultValue(TRUE);
+
+  \Drupal::entityDefinitionUpdateManager()
+    ->installFieldStorageDefinition('status', 'acm_sku', 'acm_sku', $storage_definition);
```
Do these functions throw exceptions? If yes, please catch and log as before. If no, do they write to the log?

japerry replied:
I don't think so, this is using the same pattern that Node uses in core, see  the following:
https://git.drupalcode.org/project/drupal/-/blob/8.8.x/core/modules/node/node.install#L169

malachy-mcconnell commented on modules/acm_sku/src/SKUFieldsManager.php:
```diff
@@ -188,29 +187,8 @@ public function updateFieldMetaInfo($field_code, array $field) {
     $config->setData($existing_fields)->save();
-
-    if ($apply_updates) {
-      $this->entityDefinitionUpdateManager->applyUpdates();
```
I don't understand why this code is removed.

japerry replied:
Automatic entity updates no longer exist in Drupal 8.7+, it is handled in the entity manager for save, therefore the existing code changing the entity should suffice.

malachy-mcconnell commented on modules/acm_sku/src/ProductOptionsManager.php:
```diff
@@ -130,7 +130,7 @@ public function loadProductOptionByOptionId($attribute_code, $option_id, $langco
    */
   public function createProductOptionWrapper($langcode, $option_id, $option_value, $attribute_id, $attribute_code, $weight) {
-    return $this->createProductOption();
+    return $this->createProductOption($langcode, $option_id, $option_value, $attribute_id, $attribute_code, $weight);
```
This is a bug fix, yes?

japerry replied:
Yes, that was taken from the merge done from the earlier PR. If you want this moved out we can do it, but its pretty minor so I just kept it in.

japerry committed:
PHPCS Code Cleanup.

# Conflicts:
#	modules/acm_cart/composer.json
#	modules/acm_cart/src/CartStorage.php

japerry committed:
Don't use @deprecated because its a submodule and d.o coding standards don't work with it.

japerry committed:
Include facets, remove deprecated behat interface.

japerry committed:
Update composer requirements.

# Conflicts:
#	modules/acm/composer.json
#	modules/acm_promotion/composer.json

japerry committed:
Update composer and info files to Drupal 9 and Drupal 8.7.7.

japerry commented:
This should be reopned.

japerry committed:
Add field group patch for D9 compatibility.

japerry committed:
Use field group 3.x-dev

japerry committed:
Use PCB 2.0

japerry committed:
Additional D9 pieces.

japerry committed:
Fix fatal testing syntax failures.

japerry committed:
D9 test fixes.

japerry committed:
ACM D9 Compatibility.

japerry commented:
The latest fix puts us in a good position to release for Drupal 9. There are some outstanding deprecations around REQUEST_TIME -- which I'll see if we need to resolve, but as of May 15 2020 -- Drupal core itself has pushed some of these deprecations to 9.1. One of the problems is the service will not work with our unit tests, which will require some more refactoring.

There is also an issue with facets, but since it doesn't have a release, we cannot test our plugin. I don't think thats a major issue at the moment.

japerry committed:
Restore all of travis' testing.
